pkg/scalers/datadog_scaler.go

```diff
@@ -60,17 +71,61 @@ func NewDatadogScaler(ctx context.Context, config *ScalerConfig) (Scaler, error)
 	}, nil
 }
 
+// parseAndTransformDatadogQuery checks correctness of the user query and adds rollup if not available
+func parseAndTransformDatadogQuery(q string, age int) (string, error) {
+	if !aggregator.MatchString(q) {
+		q = "avg:" + q
+	}
+
+	if !filter.MatchString(q) {
+		return "", fmt.Errorf("malformed Datadog query")
+	}
+
+	match := rollup.FindStringSubmatch(q)
```

Comment:
Would be nice to add some comment here, with an example contents that are being parsed and some description. Just for clarity.

Reply:
Added comments and added tests for that function